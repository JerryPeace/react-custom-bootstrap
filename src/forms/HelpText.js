import PropTypes from 'prop-types';
import React from 'react';
import bootstrapUtils from '../utils/bootstrapUtils';
import Glyphicon from '../glyphicon/Glyphicon'
import styleMaps from '../styleMaps';
import classNames from 'classnames';

class HelpText extends React.Component {
    static propTypes = {
        text: PropTypes.string,
        bsClass: PropTypes.string
    };

    static defaultProps = {
        bsClass: 'help-block'
    };

    render() {
        const {text, className, children, ...props} = this.props,
            bsCls = bootstrapUtils.getCls(props);

        return (
            <div className={classNames(className)}>
                <span className={classNames('msg', bsCls)}>
                    {text}
                    {children}
                </span>
            </div>
        );
    }
}

export default HelpText;
