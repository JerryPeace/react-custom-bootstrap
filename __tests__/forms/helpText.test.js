import React from 'react';
import { mount } from 'enzyme';
import HelpText from 'forms/HelpText';
import toJSON from'enzyme-to-json';

describe('HelpText', () => {

    let helpText;

    describe('HelpText', () => {

        beforeEach(() => {
            helpText = mount(<HelpText />);
        });

        test('should render corrently', () => {
            expect(toJSON(helpText)).toMatchSnapshot();
        });

    });

});
