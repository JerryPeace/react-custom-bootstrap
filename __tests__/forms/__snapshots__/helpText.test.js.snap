// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpText HelpText should render corrently 1`] = `
<HelpText
  bsClass="help-block"
>
  <div
    className=""
  >
    <span
      className="msg help-block"
    />
  </div>
</HelpText>
`;
